app/redux/ducks/interventions.js

```diff
@@ -46,12 +50,44 @@ export function unsubscribe(channel) {
   return { type: UNSUBSCRIBE, payload: channel };
 }
 
-export function notify(notification) {
-  const { message } = notification.data;
-  const { type } = message;
-  if (type === 'subject_queue') {
-    const subjectIDs = message.subject_ids;
-    const workflowID = message.workflow_id;
+export function intervention(message) {
+  // only process experiment events from sugar
+  // we could add checking for the logged in user channel (id)
+  // as well if we don't trust sugar
+  // Example payload data from sugar
+  // {
+  //  channel: "user:27"
+  //    data: {
+  //      event: "intervention",
+  //      event_type: "message",
+  //      message: "You are a star! All of your contributions really help.",
+  //      project_id: "908",
+  //      type: "experiment",
+  //      user_id: "27",
+  //    }
+  //  type: "experiment"
+  // }
+
+  var { type = 'unknown' } = message;
+  if (type !== "experiment") {
+    return { type: UNKNOWN_EXPERIMENT };
+  };
+
+  var { data = 'missing' } = message;
+  if (data === 'missing') {
+    return { type: MISSING_DATA };
+  };
+
+  var { event = 'unknown' } = data;
+  if (event !== 'intervention') {
+    return { type: UNKNOWN_EVENT };
+  };
+
+  const { event_type } = data;
+
+  if (event_type === 'subject_queue') {
```

Comment:
Should this be a switch on `event_type`? That might make it easier to add new event types in future.

Reply:
i've added the switch and extracted to method to make the code clearer.